crates/cheatcodes/src/inspector.rs

```diff
@@ -939,6 +938,13 @@ impl<DB: DatabaseExt> Inspector<DB> for Cheatcodes {
                 if data.journaled_state.depth() == broadcast.depth {
                     data.env.tx.caller = broadcast.original_origin;
 
+                    if !call.is_static {
+                        let account =
+                            data.journaled_state.state().get_mut(&broadcast.new_origin).unwrap();
+
+                        account.info.nonce += 1;
+                    }
```

Comment:
currently we are increasing nonce _before_ broadcasting CALL to simulate nonce increase during on-chain transaction

this is incorrect for`--isolate` because we need nonce to be up-to-date at the point when we are creating a transaction

so the change is to increase nonce _after_ the CALL

however, thinking of it now, with new workaround when we explicitly decrease nonces in isolation, this is not really needed as long as we touch the account when pre-increase its nonce, updated in 33814e54f4a6cc1614ed7e06f042124ab51ada0a

Reply:
yeah can we also doc that

Reply:
good call